tests/sklearn/test_sklearn_models.py

```diff
@@ -1139,6 +1162,76 @@ def check_load_fitted_sklearn_linear_models(model_class, n_bits, x, y, check_flo
     )
 
 
+def check_rounding_consistency(
+    model,
+    n_bits,
+    x,
+    y,
+    predict_method,
+    metric,
+    is_weekly_option,
+):
+    """Test that Concrete ML without and with rounding are 'equivalent'."""
+
+    # Run the test with more samples during weekly CIs
+    if is_weekly_option:
+        fhe_test = get_random_samples(x, n_sample=5)
+
+    # Check that rounding is enabled
+    rounding_enabled = os.getenv("USE_ROUNDING") == "True"
+    assert rounding_enabled
+
+    # Fit and compile with rounding enabled
+    circuit_with_rounding = fit_and_compile(model, x, y)
+
+    rounded_predict_quantized = predict_method(x, fhe="disable")
+    rounded_predict_simulate = predict_method(x, fhe="simulate")
+
+    # Compute the FHE predictions only during weekly CIs
+    if is_weekly_option:
+        rounded_predict_fhe = predict_method(fhe_test, fhe="execute")
+
+    # For models that has less than 9 bits of quantization, we repeat the same experiment, but this
+    # time with rounding disabled
+    if n_bits <= 9:
+        with pytest.MonkeyPatch.context() as mp_context:
+
+            # Disable rounding
+            mp_context.setenv("USE_ROUNDING", "False")
+
+            # Check that rounding is disabled
+            rounding_disabled = os.environ.get("USE_ROUNDING") == "False"
+            assert rounding_disabled
+
+            with pytest.warns(
+                DeprecationWarning,
+                match=(
+                    "Using Concrete tree-based models without the `rounding feature` is "
+                    "deprecated.*"
+                ),
+            ):
+
+                # Fit and compile without rounding
+                circuit_without_rounding = fit_and_compile(model, x, y)
+
+            not_rounded_predict_quantized = predict_method(x, fhe="disable")
+            not_rounded_predict_simulate = predict_method(x, fhe="simulate")
+
+        metric(rounded_predict_quantized, not_rounded_predict_quantized)
+        metric(rounded_predict_simulate, not_rounded_predict_simulate)
+
+        # Compute the FHE predictions only during weekly CIs
+        if is_weekly_option:
+            not_rounded_predict_fhe = predict_method(fhe_test, fhe="execute")
+            metric(rounded_predict_fhe, not_rounded_predict_fhe)
+
+        # Check that the maximum bitwidth of the cuircuit with rounding is at most n_bits + 2
```

Comment:
typo : `cuircuit` -> `circuit` and `bitwidth` -> `bit-width`

also, similar to what I've said right below, it's more like `Check that the maximum bit-width of the circuit with rounding is at most n_bits - 2` right ?

Reply:
No, with rounding: the subtraction: x - y cost at most 2 bits.
That's why we have 2 extra bits in the circuit

Reply:
were there nevery any flakys with +2 ? do we need +3 ?

Reply:
For the moment, i only saw +1 and +2
But never +3